test/test-csp-inline.html

```diff
@@ -0,0 +1,54 @@
+<!doctype html>
+<html>
+  <head>
+    <link rel="stylesheet" type="text/css" href="../bower_components/qunit/qunit/qunit.css"/>
+    <meta http-equiv="Content-Security-Policy" content="script-src 'self' 'nonce-abc';">
+  </head>
+  <body>
+
+    <h1 id="qunit-header">SystemJS Test Suite</h1>
+
+    <h2 id="qunit-banner"></h2>
+    <div id="qunit-testrunner-toolbar"></div>
+    <h2 id="qunit-userAgent"></h2>
+    <ol id="qunit-tests"></ol>
+    <div id="qunit-test-area"></div>
+
+    <script src="../dist/system.src.js" type="text/javascript"></script>
+    <script src="../bower_components/qunit/qunit/qunit.js"></script>
+
+    <script nonce='abc'>
+      QUnit.config.testTimeout = 2000;
+
+      QUnit.module("SystemJS nonce");
+
+      function err(e) {
+        setTimeout(function() {
+          throw e.stack || e;
+          start();
+        });
+      }
+
+      asyncTest('Importing a script with a good nonce', function() {
+        System.config({
+          meta: {
+            'tests/csp/nonce.js': {
+              nonce: 'abc'
+            }
+          }
+        });
+        System['import']('tests/csp/nonce.js').then(function(m) {
+          ok(m.nonce === 'abc');
+          start();
+        });
+      });
+
+      asyncTest('Importing a script without a nonce', function() {
+        System['import']('tests/csp/nonce.js').then(function(m) {
```

Comment:
This part is weird. When CSP in Chrome blocks a resource, it will just be an empty Object and no error is called. Ideally we could figure out how to throw an error when a resource is blocked. I assume this is to prevent brute forcing a nonce.